### PR TITLE
sample playbooks for Ibm concert

### PR DESCRIPTION
**Pull request by adn1107**

### CODE REVIEW

File: ibm_concert/cert_renewal/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+- [**print_hc_buffer**](roles/print_hc_buffer/README.md) - Pull data from Health Checker
+- [**send-template**](roles/send-template/README.md) - send template to a zOS host
+
+## Set up job templates on Ansible Automation Platform
+These playbooks are designed to be used with Ansible Automation Platform (AAP) job templates. The information in the `host_vars` can be used to set up Inventory and Hosts on AAP.
+
+Review the required inputs to each playbooks to set up Surveys on AAP so that external callers can call the AAP REST API correctly.
+
+- Set up an AAP schedule to send certificate data to IBM Concert on a regular basis
+- Set up an AAP job template to renew a certificate data to IBM Concert on a regular basis
```
Comment (itsBryantP) [inline ibm_concert/cert_renewal/README.md:43]: is this supposed to say "Set up an AAP job template to renew a certificate on a regular basis" ?

Comment (adn1107) [reply ibm_concert/cert_renewal/README.md:43]: yes

File: ibm_concert/cert_renewal/roles/send-template/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+Role Name
+=========
+
+A brief description of the role goes here.
```
Comment (itsBryantP) [inline ibm_concert/cert_renewal/roles/send-template/README.md:4]: role likely not  needed if using templating feature of zos_job_submit.

File: ibm_concert/cert_renewal/send_cert_data.yml [hunk 3]
```diff
@@ -0,0 +1,186 @@
+###############################################################################
+# © Copyright IBM Corporation 2024
+###############################################################################
+
+- hosts: all
+  collections:
```
Comment (itsBryantP) [inline ibm_concert/cert_renewal/send_cert_data.yml:6]: similar comment to another regarding using FQCN

image

**Review by itsBryantP — CHANGES_REQUESTED** [review]

File: ibm_concert/cert_renewal/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+## Set up ServiceNow for Ansible Automation Platform Integration
```
[the remark this answers is not in the document]
Comment (adn1107) [reply ibm_concert/cert_renewal/README.md:45]: No setup steps for AAP. I have setup steps for ServiceNow, but I don't know where they will be published (i.e. Concert Knowledge Center or a blog)

File: ibm_concert/cert_renewal/ansible.cfg [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+# Minimum required configuration for this playbook.
+# For additional `ansible.cfg` configuration options, review:
+#   https://docs.ansible.com/ansible/latest/reference_appendices/config.html
+#   https://github.com/ansible/ansible/blob/devel/examples/ansible.cfg
+################################################################################
+
+[defaults]
+forks = 25
+display_skipped_hosts = no
+callback_whitelist = profile_tasks, timer
```
[the remark this answers is not in the document]
Comment (adn1107) [reply ibm_concert/cert_renewal/ansible.cfg:15]: will address in a new PR

File: ibm_concert/cert_renewal/inventories/host_vars/zos_host.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+#    e.g, pyz: "/usr/lpp/IBM/cyp/v3r8/pyz"
+# - Property `ZOAU` is the ZOAU installation home on the z/OS managed node (target),
+#    e.g, zoau: "/usr/lpp/IBM/zoautil"
+# - property `ansible_python_interpreter` is the z/OS managed node (target) Python
+#    binary installation path, e.g, ansible_python_interpreter: "{{PYZ}}/bin/python3"
+#
+# Note, PYZ and ZOAU environment variables must be configured.
+################################################################################
+
+PYZ: "/usr/lpp/IBM/cyp/v3r11/pyz"
```
[the remark this answers is not in the document]
Comment (adn1107) [reply ibm_concert/cert_renewal/inventories/host_vars/zos_host.yml:18]: will address in a new PR

File: ibm_concert/cert_renewal/renew_cert.yml [hunk 7]
```diff
@@ -0,0 +1,314 @@
+###############################################################################
+# © Copyright IBM Corporation 2024
+###############################################################################
+
+- hosts: all
+  collections:
+    - ibm.ibm_zos_core
```
[the remark this answers is not in the document]
Comment (adn1107) [reply ibm_concert/cert_renewal/renew_cert.yml:7]: will address in a new PR

File: ibm_concert/cert_renewal/renew_cert.yml [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+  gather_facts: false
+  environment: "{{ environment_vars }}"
+  vars:
+    ansible_ssh_pipelining: true
```
[the remark this answers is not in the document]
Comment (adn1107) [reply ibm_concert/cert_renewal/renew_cert.yml:11]: will address in a new PR

File: ibm_concert/cert_renewal/renew_cert.yml [hunk 9]
```diff
@@ -0,0 +1,314 @@
...
+
+    - name: Create temporary directory to store bank files
+      ansible.builtin.tempfile:
+        state: directory
+        path: /tmp
+      register: playbook_tmp_dir
+
+    - block:
+        - ansible.builtin.include_role:
+            name: issue_operator_cmd
```
[the remark this answers is not in the document]
Comment (adn1107) [reply ibm_concert/cert_renewal/renew_cert.yml:79]: These roles were created in the early days of the samples to show new Ansible users on how to structure roles in playbooks. Would require rewrite and testing if we remove them now.

File: ibm_concert/cert_renewal/renew_cert.yml [hunk 10]
```diff
@@ -0,0 +1,314 @@
...
+          vars:
+            task_description: 'Run Health Checker'
+            command: "F HZSPROC,RUN,CHECK=(IBMRACF,RACF_CERTIFICATE_EXPIRATION)"
+
+        - ansible.builtin.include_role:
+            name: print_hc_buffer
+          vars:
+            hc_check: 'IBMRACF,RACF_CERTIFICATE_EXPIRATION'
```
[the remark this answers is not in the document]
Comment (adn1107) [reply ibm_concert/cert_renewal/renew_cert.yml:87]: These roles were created in the early days of the samples to show new Ansible users on how to structure roles in playbooks. Would require rewrite and testing if we remove them now.

File: ibm_concert/cert_renewal/renew_cert.yml [hunk 11]
```diff
@@ -0,0 +1,314 @@
...
+            role_cert: "{{cert_args}}"
+            tso_command:
+              - RACDCERT SITE LIST(LABEL('{{cert_args.cert_label}}'))
+          when: cert_type is defined and cert_type == 'SITE'
+
+        - debug:
+            var: cert_args
+
+        - ansible.builtin.include_role:
+            name: issue_tso_cmd
```
[the remark this answers is not in the document]
Comment (adn1107) [reply ibm_concert/cert_renewal/renew_cert.yml:167]: These roles were created in the early days of the samples to show new Ansible users on how to structure roles in playbooks. Would require rewrite and testing if we remove them now.

File: ibm_concert/cert_renewal/roles/send-template/README.md (hunk 2 above)
Comment (adn1107) [reply ibm_concert/cert_renewal/roles/send-template/README.md:4]: These roles were created in the early days of the samples to show new Ansible users on how to structure roles in playbooks. Would require rewrite and testing if we remove them now.

File: ibm_concert/cert_renewal/send_cert_data.yml (hunk 3 above)
Comment (adn1107) [reply ibm_concert/cert_renewal/send_cert_data.yml:6]: will address in a new PR

**Comment by adn1107** [comment]
Close this PR. Will open new one to address the PR requests.

**Pull request closed without merging**